test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -2649,8 +2742,29 @@ def cgroup_offline_node(self, name, vnpernuma=False):
             self.skipTest('pbs_cgroups_hook: Failed to copy '
                           'freezer state FROZEN')
 
-        # Make sure the kernel knows about the freeze on MoM
-        time.sleep(1)
+        confirmed_frozen = False
+
+        for count in range(30):
+            ret = self.du.cat(hostname=self.hosts_list[0],
+                              filename=freezer_file,
+                              sudo=True)
+            if ret['rc'] != 0:
+                self.logger.info("Cannot confirm freezer state"
+                                 "sleeping 30 seconds instead")
+                confirmed_frozen = True
+                time.sleep(30)
+                break
+            if ret['out'][0] == 'FROZEN':
+                self.logger.info("job processes reported as FROZEN")
+                confirmed_frozen = True
+                break
+            else:
+                self.logger.info("freezer state reported as "
+                                 + ret['out'][0])
+                time.sleep(1)
+
+        if not confirmed_frozen:
+            self.logger.info("Freezer did not work -- skip test")
```

Comment:
So when this is reached, somewhere along the line, the test will get skipped? There's no need for a self.skip_test(...) call?

Reply:
As said before -- we MUST NOT just do a skipTest unless we're sure the freezer created is thawed.

Reply:
Got it.